### ISSUE TITLE
Program not working with non-example images

### ISSUE THREAD

**Issue by tovvvar** [sparkfun/BMPtoArray#1]
The python script works perfectly with the example bitmaps given, but fails when I try to use any other images. 

Python27>python.exe bmp2array.py che.bmp
Size of file: 6198
Offset: 1078
Number of colors used: 256
output.txt complete
Copy and paste this array into a image.h or other header file

Python27>python.exe bmp2array.py test.bmp
Size of file: 15498
Offset: 138
Number of colors used: 0
Traceback (most recent call last):
  File "bmp2array.py", line 80, in <module>
    actualColor1 = colorIndex[colorCode1] #Look up this code in the table
IndexError: bytearray index out of range

The above error is what I get when using other bitmaps of exactly the same size. 
Please let me know what that error might suggest, or what the table is that the code is referencing, or if it might just be a problem with creating new bitmaps (the file size seems to be larger despite being the same size). 
This would be super useful if I can get it to work!

**Comment by nseidle** [comment]
Thanks for posting the output! For some reason your bmp is coming back with 0 colors used. Can you post your BMP or a similar BMP that fails?

**Comment by tovvvar** [comment]
[BMPtest.bmp.zip](https://github.com/sparkfun/BMPtoArray/files/2138009/BMPtest.bmp.zip)

This and several other BMPs were converted from JPGs using this site - https://image.online-convert.com/convert-to-bmp
I also tried a BMP created in Photoshop and that failed as well.
How were the example BMPs created?

**Comment by nseidle** [comment]
Ok, I opened your landscape bmp (very cool by the way, can't wait to see what it looks like on our OLED! please post photos!) and converted the BMP to 'Black and White'. This takes the bmp from 16kb to 7kb. It retains the grayscale, but reduces the number of colors to 256. The python script then completes successfully.

I use paint.net on Windows but use whatever you need.

Attached is a zip with your original (doesn't work), with blackandwhite coverted (works) bmp, and output.

[landscape.zip](https://github.com/sparkfun/BMPtoArray/files/2138112/landscape.zip)

This is very much a limitation of my python script, but I think it's best to leave the pre-processing of the bitmap to heftier image editing programs. I'll add this info to the readme to help future users.

**Comment by tovvvar** [comment]
Perfect, thanks for the help! What did you use to convert the image to black and white, just so I know exactly what to do for future images?

**Comment by nseidle** [comment]
Np! I used [paint.net](https://www.getpaint.net/) in Windows. It works well for what I need (basic resize and stuff). I bet some of the [online image converters](https://www.google.com/search?q=convert+bmp+to+black+and+white) would do the same.

**Comment by Rogefeller** [comment]
Just making a note that I had the same issue. I could not get this script to work with non-example images. Using both MSPaint and Paint Shop Pro, I was trying every BMP format & encoding option available and could not get anything to work. . I was about to switch to Photoshop until I saw this issue note.

Paint.net worked.

I would suggest just stating to use paint.net in the Readme file because the convert JPEG to BMP, then downgrade to grayscale, is not sufficient when using other image editing software.

Another thing worth mentioning is to change the resolution to 96 pixels/inch. Script did not work with any other resolution. 

Great script once I got a compatible BMP file!
